### PR TITLE
Title Change: add a rocblas transpose kernel 

### PR DESCRIPTION
**Pull request by tingxingdong**
-
add a transpose kernel in library and transpose & bandwidth test in client. 
this pull request has nothing to do with boost cmake now.

### CODE REVIEW

File: clients/benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -77,6 +77,7 @@ target_include_directories( client
     $<BUILD_INTERFACE:${Boost_INCLUDE_DIRS}>
 )
 
+link_libraries( ${BOOST_ROOT}/lib )
```
Comment (kknox) [inline clients/benchmarks/CMakeLists.txt]: I don't think you need to use link_libraries()?  ${Boost_LIBRARIES} should be absolute paths.  Typically, explicitly calling link_libraries is a bad idea.

Comment (tingxingdong) [inline clients/benchmarks/CMakeLists.txt]: NO, It does not give absolutely paths.  If invoking the executable, it fails. 

/home/tim/usr/cget/pkg/kitware__cmake/install/bin/cmake -E cmake_link_script CMakeFiles/client.dir/link.txt --verbose=1
/opt/rocm/hip/bin/hipcc    -m64 -O3 -DNDEBUG   CMakeFiles/client.dir/client.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/utility.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/cblas_interface.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/flops.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/norm.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/unit.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/near.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/arg_check.cpp.o CMakeFiles/client.dir/home/tim/rocBLAS_tindong/clients/common/rocblas_template_specialization.cpp.o  -o client -rdynamic /usr/local/lib/librocblas-hcc.a **-lboost_program_options** -lblas -llapack -lpthread /usr/local/lib/libtensile-rocblas.a

Comment (kknox) [inline clients/benchmarks/CMakeLists.txt]: `/home/tim/usr/ **cget** /pkg/kitware__cmake/install/bin/cmake`

Are you using Paul's cget?

Comment (tingxingdong) [inline clients/benchmarks/CMakeLists.txt]: Long time ago, Paul installed cget in order to get the latest cmake on my machine

Comment (kknox) [inline clients/benchmarks/CMakeLists.txt]: I'm looking into this; i don't want to use link_libraries(), which would cause problems down the road.

**Comment by kknox** [comment]
Tim, this looks to be legit failures in unit tests (core dump).

**Comment by kknox** [comment]
I think this PR is dead; shall we close it out?

**Pull request closed without merging**